andela/ah-commando-backend
# #167190452 adds search functionality

#### What does this PR do?
- It allows a user to search for an article using a keyword.
- once the user gets the search result, the user can filter the result gotten from the search

#### Description of Task to be completed?
- added the route `GET /api/v1/articles/searchQuery={keyword}`
- added the route `POST /api/v1/articles/search/filter/searchQuery={keyword}` 

#### How should this be manually tested?
- fetch or pull this branch
- run `npm install` to get all required packages
- visit `GET /api/v1/articles?searchQuery={keyword}` with a key, you would see articles related to that keyword by article title, description, authors name and username.
- you can add optional page and/or limit to the url. For example `GET /api/v1/articles?searchQuery=mon&limit=10&page=1`

- visit `POST /api/v1/articles/search/filter?searchQuery={keyword}` with a list of categories, author names or tags (separated by a coma) or space or a combination of any to the filter including the keyword to search for articles.
```
   Example of request body.
  {
    categories: 'technology,health',
    authorNames: 'peter'
    tags:'javascript'
  }
```
#### Any background context you want to provide?
No

#### What are the relevant pivotal tracker stories?
[#167190452](https://www.pivotaltracker.com/story/show/167190452)

#### Screenshots (if appropriate)
<img width="858" alt="Screenshot 2019-08-06 at 3 57 56 PM" src="https://user-images.githubusercontent.com/37886814/62551198-7a060a00-b863-11e9-90a4-ac5ba3f62338.png">
<img width="782" alt="Screenshot 2019-08-06 at 3 58 13 PM" src="https://user-images.githubusercontent.com/37886814/62551199-7a060a00-b863-11e9-9c9c-0b1ce4488353.png">
<img width="953" alt="Screenshot 2019-08-06 at 3 59 36 PM" src="https://user-images.githubusercontent.com/37886814/62551205-7c686400-b863-11e9-8ad6-dd1fc2350542.png">
<img width="825" alt="Screenshot 2019-08-06 at 3 59 59 PM" src="https://user-images.githubusercontent.com/37886814/62551222-84280880-b863-11e9-90ec-897e6bf28d67.png">
<img width="848" alt="Screenshot 2019-08-07 at 3 51 11 PM" src="https://user-images.githubusercontent.com/37886814/62633843-b94c5d80-b92c-11e9-8f81-91af54b0d182.png">

#### Questions:
None
